src/main/scala/diplomaticobjectmodel/logicaltree/LogicalTrees.scala

```diff
@@ -97,3 +99,71 @@ class SubSystemLogicalTreeNode(var getOMInterruptDevice: (ResourceBindingsMap) =
     )
   }
 }
+
+case class OMSRAMData[T <: Data](
+  data: T,
+  resourceBindings: ResourceBindings,
+  architecture: RAMArchitecture,
+  description: String,
+  depth: Int,
+  ecc: Option[OMECC] = None,
+  hasAtomics: Option[Boolean] = None
+)
+
+class SRAMLogicalTreeNode[T <: Data](description: String, architecture: RAMArchitecture, depth: Int, lanes: Int, bits: Int, device: () => Device,
+  rbm: () => ResourceBindingsMap,
+  ecc: Option[OMECC] = None,
+  hasAtomics: Option[Boolean] = None) extends LogicalTreeNode {
+  def getOMComponents(resourceBindingsMap: ResourceBindingsMap, components: Seq[OMComponent]): Seq[OMComponent] = {
+    Seq(
+      DiplomaticObjectModelAddressing.makeOMMemory(
+        desc = description,
+        depth = depth,
+        data = Vec(lanes, UInt(width = bits)),
+        resourceBindings = Some(rbm().map(device())),
+        architecture = Some(architecture),
+        ecc = ecc,
+        hasAtomics = hasAtomics
+      )
+    )
+  }
+}
+
+class ImpLogicalTreeNode extends LogicalTreeNode {
+  override def getOMComponents(resourceBindingsMap: ResourceBindingsMap, children: Seq[OMComponent] = Nil): Seq[OMComponent] =
+    children
+}
+
+class TestSoCLogicalTreeNode extends LogicalTreeNode {
+  override def getOMComponents(resourceBindingsMap: ResourceBindingsMap, children: Seq[OMComponent] = Nil): Seq[OMComponent] = {
+    Seq(OMTestHarness( components = children))
+  }
+}
+
+case object DeviceKey extends Field[Option[Device]](
+  None
+)
+
+case object ParentLogicalTreeNodeKey extends Field[Option[LogicalTreeNode]](
+  None
+)
+
+case object ResourceBindingsMapLogicalTreeNodeKey extends Field[Option[ResourceBindingsMap]](
```

Comment:
What does this have to do with "LogicalTreeNodes"? Also, why do you need it? Aren't the ResourceBindings supposed to be resolved when you walk the tree at the end?